src/box/memtx_tree.cc

```diff
@@ -782,21 +783,24 @@ prepare_start_prefix_iterator(struct memtx_tree_key_data<USE_HINT> *start_data,
  * @param type - the lookup iterator type, may be updated;
  * @param region - the region to allocate a new @a start_data on if required.
  * @param[out] iterator - the result of the lookup;
+ * @param[out] offset - if NEED_OFFSET, the offset @a iterator points to,
+ *  undefined value otherwise;
  * @param[out] equals - true if the lookup gave the exact key match;
- * @param[out] initial_elem - the element approached on initial lookup, stepped
- *  over if the iterator is reverse, see the end of this function for more info.
+ * @param[out] initial_elem - the optional pointer to the element approached on
+ *  the initial lookup, stepped over if the iterator is reverse, see the end of
+ *  this function for more information.
  *
  * @retval true on success;
  * @retval false if the iteration must be stopped without an error.
  */
-template<bool USE_HINT>
+template<bool USE_HINT, bool NEED_ITERATOR = true, bool NEED_OFFSET = false>
```

Comment:
It's okay to always return offset and iterator. It's cheap. If they aren't needed, they will be optimized out by the compiler. Please drop the template parameters.

Reply:
Done.